### PR TITLE
Add NOTICES.txt to the project

### PR DESCRIPTION
**Pull request by izgeri**
#### What does this PR do (include background context, if relevant)?
Adds a NOTICES.txt to the project with OSS dependency license information and some utilities for tracking dependency changes. At this stage, the dependency license tracking is still largely manual, but ideally we can continue to progress to automated dependency license tracking pre-releases.

#### What ticket does this PR close?
Connected to #727

### CODE REVIEW

File: bin/check_dependencies [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+set -uo pipefail
```
Comment (sgnn7) [inline bin/check_dependencies]: Why no `-e`?

File: bin/check_dependencies [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# get a new list of project licenses
+# this command syntax is wild but it's what the license_finder image requires
+docker run --rm \
+  -v $toplevel_dir:/scan \
+  licensefinder/license_finder \
+  /bin/bash -lc "cd /scan && license_finder" \
+  > $toplevel_dir/assets/updated_licenses.txt
+
+# check if there are any differences; if not, exit
+diff=$(diff $toplevel_dir/assets/updated_licenses.txt $toplevel_dir/assets/license_finder.txt)
```
Comment (sgnn7) [inline bin/check_dependencies]: Is diff common enough to expect it? If so, ignore the comment.

File: bin/check_dependencies [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+if [ -z "$diff" ]; then
+  echo "No dependencies have changed."
+else
+  echo -e "\nWarning - dependencies have changed. NOTICES.txt will need to be updated.\n"
+  echo "$diff"
+fi
```
Comment (sgnn7) [inline bin/check_dependencies]: I don't see any error-type exits here - I'd assume that we eventually want this automated so the exit code of "1" would go a long way to helping automate this even in a simple checking script on Jenkins.

File: NOTICES.txt [hunk 4]
```diff
@@ -0,0 +1,1494 @@
+
```
Comment (sgnn7) [inline NOTICES.txt]: After re-reading this, I think we really need something better than this and split up into parts.

**Comment by sgnn7** [comment]
PS: CodeClimate is also complaining about something.

File: bin/check_dependencies (hunk 1 above)
Comment (izgeri) [reply bin/check_dependencies]: With it the script would die after the docker run or after diff

File: NOTICES.txt (hunk 4 above)
Comment (izgeri) [reply NOTICES.txt]: I based this on the doc that Pivotal requires for us, and I think it's much easier to read than the typical one our org uses. What specifically would you suggest that we change?

File: bin/check_dependencies (hunk 2 above)
Comment (izgeri) [reply bin/check_dependencies]: For maintainers? I think so

File: bin/check_dependencies (hunk 3 above)
Comment (izgeri) [reply bin/check_dependencies]: This isn't currently part of the pipeline, I don't want it to be until the end to end of generating the file is actually automated, so for now this is a util to check if the txt needs to be updated pre-release.

I'm fine with maybe adding those in later, but right now this is manual and informational.

**Comment by izgeri** [comment]
@sgnn7 I fixed the code climate issues, this should be gtg now - but some of your comments I responded to rather than changing what I did. lmk what you think.

File: bin/check_dependencies (hunk 3 above)
Comment (sgnn7) [reply bin/check_dependencies]: Can you just add a comment or a TODO on this for that then? I'd hate for it to be lost in this PR and never fixed up later.

**Pull request merged**